src/py-opentimelineio/opentimelineio/adapters/svg.py

```diff
@@ -0,0 +1,1017 @@
+}
+
+random_colors_used = []
+
+
+def get_random_color():
+    return [random(), random(), random()]
+
+
+def color_distance(c1, c2):
+    return sum([abs(x[0] - x[1]) for x in zip(c1, c2)])
+
+
+def generate_new_color(existing_colors):
+    max_distance = None
+    best_color = None
+    for i in range(0, 100):
+        color = get_random_color()
+        if not existing_colors:
+            return color
+        best_distance = min([color_distance(color, c) for c in existing_colors])
+        if not max_distance or best_distance > max_distance:
+            max_distance = best_distance
+            best_color = color
+    return best_color
+
+
+def random_color():
+    color = generate_new_color(random_colors_used)
+    random_colors_used.append(color)
+    return color[0] * 255.0, color[1] * 255.0, color[2] * 255.0, 255.0
+
+
+Point = namedtuple('Point', ['x', 'y'])
+
+
+class Rect(object):
+    origin = Point(0, 0)
+    width = 0.0
+    height = 0.0
+
+    def __init__(self, origin=Point(0, 0), width=0.0, height=0.0):
+        self.origin = origin
+        self.width = width
+        self.height = height
+
+    def normalized(self):
+        normalized_origin = Point(
+            self.origin.x + (self.width if self.width < 0 else 0),
+            self.origin.y + (self.height if self.height < 0 else 0),
+        )
+        normalized_width = abs(self.width)
+        normalized_height = abs(self.height)
+        return Rect(normalized_origin, normalized_width, normalized_height)
+
+    def min_x(self):
+        return self.normalized().origin.x
+
+    def min_y(self):
+        return self.normalized().origin.y
+
+    def mid_x(self):
+        return self.origin.x + (self.width * 0.5)
+
+    def mid_y(self):
+        return self.origin.y + (self.height * 0.5)
+
+    def max_x(self):
+        norm = self.normalized()
+        return norm.origin.x + norm.width
+
+    def max_y(self):
+        norm = self.normalized()
+        return norm.origin.y + norm.height
+
+    def contract(self, distance):
+        self.origin.x += distance
+        self.origin.y += distance
+        self.width -= 2.0 * distance
+        self.height -= 2.0 * distance
+
+
+LCARS_CHAR_SIZE_ARRAY = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                         0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                         0, 0, 17, 26, 46, 63, 42, 105, 45, 20, 25, 25, 47, 39, 21,
+                         34, 26, 36, 36, 28, 36, 36, 36,
+                         36, 36, 36, 36, 36, 27, 27, 36, 35, 36, 35, 65, 42, 43, 42,
+                         44, 35, 34, 43, 46, 25, 39, 40,
+                         31, 59, 47, 43, 41, 43, 44, 39, 28, 44, 43, 65, 37, 39, 34,
+                         37, 42, 37, 50, 37, 32, 43, 43,
+                         39, 43, 40, 30, 42, 45, 23, 25, 39, 23, 67, 45, 41, 43, 42,
+                         30, 40, 28, 45, 33, 52, 33, 36,
+                         31, 39, 26, 39, 55]
```

Comment:
This is very brittle, if the font gets changed for whatever reason, this array will become invalid. This also will break for characters outside the ASCII range, like é. That said, I'm not sure if there is a smarter library for querying this info in the standard library.
If we are hard-coding font info, it may be useful to create something like a `Font` class that bundles together this data.

It could look like:

```python
class Font:
    def __init__(self, name, character_widths):
        self.name = name
        self._character_widths = character_widths

    def width_for_character(self, char):
        try:
            return self._character_widths[ord(char)]
        except IndexError:
            raise ValueError("Unsupported character {}".format(char))

ROBOTO = Font("Roboto", ...char width list here...)
```

This abstracts the behavior in a way where future people can either implement alternate fonts, or add in usage for a library that can provide the important information.

Reply:
I would strongly recommend not introducing any dependencies for font handling because there aren't really any lightweight libraries that I'm aware of, at all. They all chain react in a lot of secondary modules, and I'm not thrilled imagining a worst case scenario of dependabot noticing a dozen new minor dependency changes every week.

If it's possible to eliminate the need to measure text at all, that would be ideal.

Reply:
If the font measurement is being used for text layout, it looks like SVG has ways to control this without having to explicitly know the font metrics: https://stackoverflow.com/questions/5546346/how-to-place-and-center-text-in-an-svg-rectangle

Reply:
I've removed this font metrics stuff. Please take a look. The tests pass on my end. Can you please rerun the actions?

PS: I'm working on all the comments, one by one.

Reply:
Relaunched action, seems to be successful again. Looks like there was a transient 504 fetching stuff from pypi